subprojects/execution/src/main/java/org/gradle/internal/execution/impl/OutputsCleaner.java

```diff
@@ -86,9 +95,12 @@ private void markDirForDeletion(@Nullable File dir) {
         }
     }
 
-    private void cleanupDirectories() throws IOException {
+    public void cleanupDirectories() throws IOException {
```

Comment:
Why do these all need to be `public` now?

Reply:
They are used from `SimpleStaleClassCleaner`.

Reply:
It seems that there should be a better way to handle this, but I don't feel strongly enough to hold up the merging of this PR.